woocommerce/woocommerce-ios
# Dynamic Dashboard: Tracking events M1

Closes: #12877 


## Description


Adds tracking events for Dynamic Dashboard M1 cards.

This PR adds the new tracking event `dynamic_dashboard_card_interacted ` that we introduced during M2 cards development to M1 cards for better consistency. 

## Testing instructions


- Create a JN site
- Create products and orders
- Login into the app using the JN store

Store onboarding
- Tap any of the tasks and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `store_setup`. 
- Navigate back and tap "View all tasks" button.
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `store_setup`. 

Performance
- Add "Performance" card to the dashboard
- Tap "View all store analytics" button 
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `performance`. 
- Switch time range and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `performance`. 
- Interact with the performance chart and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `performance`. 

Top performers
- Add "Top performers" card to the dashboard
- Tap "View all store analytics" button 
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `top_performers`. 
- Switch time range and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `top_performers`. 
- Tap on a top-performing product and confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `top_performers`. 

Blaze campaigns
- Add "Blaze campaigns" card to the dashboard
- Tap "View all campaigns" button 
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `blaze`. 
- Interact with the Blaze campaign card (tap on a product or campaign)
- Confirm that `dynamic_dashboard_card_interacted` event is tracked with `type` as `blaze`. 

## Screenshots


| Store onboarding | Performance | Top performers | Blaze campaigns |
|--------|--------|--------|--------|
| <img width="345" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/b779f279-25ad-4fd5-a672-efde3799fd5f"> | <img width="345" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/12ba9a21-ca19-4f69-bb5b-7836d673a22d"> | <img width="349" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/27c64490-1b9e-4291-9a10-f8bb836e9303"> | <img width="350" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/56889cc3-2831-4133-bfa4-0dbb545a5918"> | 

---
